joplin/api/schema.py

```diff
@@ -6,19 +6,116 @@
+
+    @staticmethod
+    def serialize(value):
+        return expand_db_html(value)
+
+
+@convert_django_field.register(RichTextField)
+def convert_rich_text_field(field, registry=None):
+    return RichTextFieldType(
+        description=field.help_text, required=not field.null
+    )
+
+
+def find_rich_text_names(StreamValue):
+    """
+    returns a list of keys of StreamField blocks that are of RichText type
+    these are the ones we need to expand_db_html on
+    TODO: this dosen't actually get us all the values we need to modify 
+    """
+    rich_text_names = []
+    rich_text_names.append('options')
+    for item in StreamValue:
+        block = item.block
+        try:
+            if isinstance(block, RichTextBlock):
+                rich_text_names.append(block.name)
+            elif block.child_blocks:
+                for child_block in block.all_blocks():
+                    if isinstance(child_block, RichTextBlock):
+                        rich_text_names.append(child_block.name)
+        except AttributeError as e:
+            pass
+    return rich_text_names
 
 
 class StreamFieldType(Scalar):
+    """
+    todo
+            for item in serialized:
+                loop through all the values and try running expand_db_html on them to return an href
+                still need to find a way to actually attach the janis_url to the selection tho..
+    """
+
     @staticmethod
-    def serialize(dt):
-        return [{'type': item.block_type, 'value': item.block.get_api_representation(item.value), 'id': item.id} for item in dt]
+    def serialize(StreamValue):
+        """
+        This is a rats nest of loops and conditionals, but the general idea is two functions:
+            1)loop through all the objects in the StreamValue, and
+                if it is a RichTextBlock,
+                add its name to a list of keys
+                ( this dosen't quite capture all the use cases as needed so it needs work)
+            2) use each of the above names as keys to sort through the stream_data
+                (which is a python dict represention of the StreamField, alot like JSON)
+                use the nested_lookup to recursivley return a list of all values that match the key
+                run a nested_alter using expand_db_html as a callback to edit those values in stream_data
+                (otherwise the values tend to get overwritten)
+
+        """
+        # get lists of potential keys
+        try:
+            rich_text_names = find_rich_text_names(StreamValue)
+        except Exception as e:
+            print(e)
+            print(traceback.format_exc())
+            pass
+
+            # todo: make things less nested
+        for block_key in rich_text_names:
+            if not isinstance(StreamValue, str):
+                data = StreamValue.stream_data
+            else:
+                data = ''
+            # sometimes data is just a string, and there isno stream_data
```

Comment:
This reads better to me above block getting data from StreamValue, as it seems to apply to both that and the check below